crates/holochain/src/core/workflow/app_validation_workflow/run_validation_callback_tests.rs

```diff
@@ -342,6 +353,85 @@ async fn validation_callback_awaiting_deps_agent_activity() {
     assert_matches!(outcome, Outcome::Accepted);
 }
 
+// An op under validation that depends on an invalid op should be rejected.
+#[tokio::test(flavor = "multi_thread")]
+async fn validation_callback_rejects_op_depending_on_invalid_op() {
```

Comment:
This tests that an op that depends on an invalid op with a `must_get_valid_record` is also rejected by app validation.